### PR TITLE
[SYCL][Graph] Update begin_recording and end_recording

### PR DESCRIPTION
**Pull request by fabiomestre**
- Removes bool return value from begin_recording() and end_recording()
- Adds missing property_list parameter to the implementation of begin_recording()

The bool return value was removed because there are no good usecases for it. The sycl way to handle errors is to return exceptions. The original reasoning to have a bool return value was that the application would likely ignore the exceptions in this entrypoint. There is no reason to believe that they will check the bool return either if it serves no practical purpose.

This is a breaking API change, but as the extension is in experimental status this is permissible.

### CODE REVIEW

**Comment by Bensuo** [comment]
There are unit tests that check the return value of begin/end recording, presumably those need updated as well?

**Comment by fabiomestre** [comment]
> There are unit tests that check the return value of begin/end recording, presumably those need updated as well?

Yeah, I forgot to run those locally. Will update / delete them.

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc [hunk 1]
```diff
@@ -1245,20 +1237,17 @@ Exceptions:
 |
 [source, c++]
 ----
-bool end_recording(const std::vector<queue>& recordingQueues)
+void end_recording(const std::vector<queue>& recordingQueues)
 ----
 
 |Synchronously changes the state of each queue in `recordingQueues` to the
-`queue_state::executing` state.
+`queue_state::executing` state. This operation is a no-op for any queue in `recordingQueues` that is already in the `queue_state::executing` state.
```
Comment (Bensuo) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc:1244]: This line should be reflowed to be less than 80 cols.

File: sycl/source/detail/graph_impl.cpp [hunk 2]
```diff
@@ -1551,56 +1554,46 @@ bool modifiable_command_graph::begin_recording(queue &RecordingQueue) {
...
-  return false;
 }
 
-bool modifiable_command_graph::begin_recording(
-    const std::vector<queue> &RecordingQueues) {
-  bool QueueStateChanged = false;
+void modifiable_command_graph::begin_recording(
+    const std::vector<queue> &RecordingQueues,
+    const sycl::property_list &PropList) {
   for (queue Queue : RecordingQueues) {
```
Comment (Bensuo) [inline sycl/source/detail/graph_impl.cpp:1568]:
```suggestion
  for (queue &Queue : RecordingQueues) {
```
Not part of this PR technically but since we're here anyway this should probably be a reference.

**Review by Bensuo — APPROVED** [review]

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc
Comment (EwanC) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc:1]: Should add yourself to the contributors section of this doc

**Review by reble — APPROVED** [review]
LGTM!

File: sycl/source/detail/graph_impl.cpp (hunk 2 above)
Comment (fabiomestre) [reply sycl/source/detail/graph_impl.cpp:1568]: As discussed in the meeting, this change cannot be done because it would result in a compiler error due to discarding the const value from the queue elements.

**Review by EwanC — APPROVED** [review]

**Review by cperkinsintel — APPROVED** [review]

**Comment by EwanC** [comment]
@intel/llvm-gatekeepers I think this is good to merge
